sdk/src/types/output.ts

```diff
@@ -77,3 +80,36 @@ export type ApplyNestedQueryFields<Schema extends object, Collection, Fields> =
  * Carry nullability of
  */
 export type RelationNullable<Relation, Output> = IsNullable<Relation, Output | null, Output>;
+
+/**
+ * Map literal types to actual output types
+ */
+export type MapFlatFields<
+	Item extends object,
+	Fields extends keyof Item,
+	FunctionMap extends Record<string, string>
+> = {
+	[F in Fields as F extends keyof FunctionMap ? FunctionMap[F] : F]: Extract<
+		Item[F],
+		keyof FieldOutputMap
+	> extends infer A
+		? A[] extends never[]
+			? Item[F]
+			: A extends keyof FieldOutputMap
+			? FieldOutputMap[A] | Exclude<Item[F], A>
+			: Item[F]
+		: Item[F];
+};
+
+// Possible JSON types
+type JsonPrimitive = null | boolean | number | string;
+type JsonValue = JsonPrimitive | JsonPrimitive[] | { [key: string]: JsonValue };
+
+/**
+ * Output map for specific literal types
+ */
+export type FieldOutputMap = {
+	json: JsonValue | null;
```

Comment:
`null` is already part of `JsonPrimitive`
```suggestion
	json: JsonValue;
```